jina/hubble/hubio.py

```diff
@@ -588,11 +587,23 @@ def fetch_meta(
             payload['tag'] = tag
 
         req_header = get_request_header()
-        resp = requests.post(pull_url, json=payload, headers=req_header)
-        if resp.status_code != 200:
-            if resp.text:
-                raise Exception(resp.text)
-            resp.raise_for_status()
+
+        logger = JinaLogger('HubIO.fetch_meta')
+        for i in range(3):
+            try:
+                resp = requests.post(pull_url, json=payload, headers=req_header)
+                if resp.status_code != 200:
+                    if resp.text:
+                        raise Exception(resp.text)
+                    resp.raise_for_status()
+                break
+            except Exception as e:
+                logger.debug(
+                    f'Fetching meta info from Jina Hub failed, retry attempt {i+1}/3. '
+                    f'Error: {e!r}'
+                )
+                if i == 2:
```

Comment:
let's make a variable NUM_RETRY

Reply:
Created `retry` decorator in `helpers` with `num_retry` argument.